nilearn/decomposition/dict_learning.py

```diff
@@ -0,0 +1,171 @@
+        means more memory for caching.
+
+    n_jobs: integer, optional, default=1
+        The number of CPUs to use to do the computation. -1 means
+        'all CPUs', -2 'all CPUs but one', and so on.
+
+    verbose: integer, optional
+        Indicate the level of verbosity. By default, nothing is printed
+
+    References
+    ----------
+
+    """
+
+    def __init__(self, mask=None, n_components=20,
+                 smoothing_fwhm=6,
+                 standardize=True,
+                 random_state=0,
+                 target_affine=None, target_shape=None,
+                 low_pass=None, high_pass=None, t_r=None,
+                 alpha=1,
+                 n_iter=1000,
+                 # Common options
+                 memory=Memory(cachedir=None), memory_level=0,
+                 n_jobs=1, verbose=0,
+                 ):
+        CanICA.__init__(self,
+                        mask=mask, memory=memory, memory_level=memory_level,
+                        n_jobs=n_jobs, verbose=verbose, do_cca=True,
+                        threshold=float(n_components), n_init=1,
+                        n_components=n_components, smoothing_fwhm=smoothing_fwhm,
+                        target_affine=target_affine, target_shape=target_shape,
+                        random_state=random_state, high_pass=high_pass, low_pass=low_pass,
+                        t_r=t_r,
+                        standardize=standardize)
+        self._keep_data_mem = True
+        # Setting n_jobs = 1 as it is slower otherwise
+        MiniBatchDictionaryLearning.__init__(self, n_components=n_components, alpha=alpha,
+                                             n_iter=n_iter, batch_size=10,
+                                             fit_algorithm='lars',
+                                             transform_algorithm='lasso_lars',
+                                             transform_alpha=alpha,
+                                             verbose=verbose,
+                                             random_state=random_state,
+                                             shuffle=True,
+                                             n_jobs=1)
+
+    def _init_dict(self, imgs, y=None, confounds=None):
+        CanICA.fit(self, imgs, y=y, confounds=confounds)
+        if isinstance(self.data_flat_, tuple):  # several subjects
+            self.data_flat_ = np.concatenate(self.data_flat_, axis=0)
+        if self.n_iter == 'auto':
+            # ceil(self.data_fat.shape[0] / self.batch_size)
+            self.n_iter = (self.data_flat_.shape[0] - 1) / self.batch_size + 1
+        if self.verbose:
+            print('[DictLearning] Learning time serie')
+        ridge = Ridge(alpha=1e-6, fit_intercept=None)
+        ridge.fit(self.components_.T, self.data_flat_.T)
+        self.dict_init = ridge.coef_.T
+        S = np.sqrt(np.sum(self.dict_init ** 2, axis=0))
+        self.dict_init /= S[np.newaxis, :]
```

Comment:
All the attributes that are estimated from data should end with an underscore: self.dict_init_ (scikit-learn convention).